gammapy/maps/region/ndmap.py

```diff
@@ -82,6 +84,8 @@ def plot(self, ax=None, axis_name=None, **kwargs):
                 )
 
         axis = self.geom.axes[axis_name]
+        if isinstance(axis, TimeMapAxis):
+            axis.time_format = time_format
```

Comment:
This will change the time format permanently on the object, not just for plotting. I'll make a suggestion in a different comment...

Reply:
I see, but isn't it better then to just modify the solution we currently have? So we could change the `time_format` from as class attribute to an instance attribute. This way use can more easily change the time format between to plots. We could also implement a setter to check for the valid formats...
 
I think the implicit change of `MapAxis.time_format` by calling `.plot()` is not a good idea. Then we should rather use the explicit solution:

```
m.geom.axes["time"].time_format = "mjd"
m.plot(axis_name="time")

# instead of

m.plot(axis_name="time", time_format="mjd")
 
```
If we prefer the latter solution in terms of API we have to choose the solution you propose and add the support to all plot methods...